ledger/partial/ptrie/partialTrie.go

```diff
@@ -31,16 +31,15 @@ func (p *PSMT) RootHash() ledger.RootHash {
 // TODO return list of indecies instead of paths
 func (p *PSMT) Get(paths []ledger.Path) ([]*ledger.Payload, error) {
 	var failedPaths []ledger.Path
-	payloads := make([]*ledger.Payload, 0)
-	for _, path := range paths {
+	payloads := make([]*ledger.Payload, len(paths))
+	for i, path := range paths {
 		// lookup the path for the payload
 		node, found := p.pathLookUp[path]
 		if !found {
-			payloads = append(payloads, nil)
```

Comment:
this changes the logic, we expect a nil value being return for not found case :-?

Reply:
Line 39 isn't necessary because `payloads` is preallocated with all elements as `nil` (`*ledger.Payload`).  

Also, for not found case `len(failedPaths) > 0`,  `ErrMissingPath` is returned with `failedPaths` and `payloads` is not used.

Reply:
right, sorry didn't pay enough attention to it.